spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/StackdriverJsonLayout.java

```diff
@@ -281,4 +281,10 @@ private void addTraceId(ILoggingEvent event, Map<String, Object> map) {
 
 		add(StackdriverTraceConstants.TRACE_ID_ATTRIBUTE, this.includeTraceId, traceId, map);
 	}
+
+	static class EnhancedLoggingEventException extends RuntimeException {
```

Comment:
Should it be `public`?

Reply:
I don't think so - it's not instantiated anywhere else and I wouldn't expect any user to create one.

Reply:
What about try-catch in user code?

Reply:
I don't think we want this exception exposed as part of our API. If we do, then it should go into its own class.

Reply:
Right, otherwise I don't see the benefit of introducing this exception class.

Reply:
That's fine with me - I put this small PR together to spur the discussion, since it could be as small as 6LOC but as @dzou noted it's probably not worth a whole class.

Reply:
I would make it its own public class. Yes, it will be a tiny one, but that's not dissimilar to marker interfaces. If the goal is to allow downstream classes to handle this exception* separately from other library-produced exceptions, then it has to officially be part of the public API.

* granted, I just "made it" a goal in the previous comment. Perhaps that's not a good goal? I think it is, but that's a "reasonable people can disagree" kind of thing.